pypesto/collections/collections.py

```diff
@@ -0,0 +1,497 @@
+            # use some short hand
+            current_cond = self.prediction_results[0].conditions[i_cond]
+
+            # create a temporary array with all the outputs needed and wanted
+            tmp_output = _stack_outputs(i_cond)
+            tmp_output_sensi = _stack_outputs_sensi(i_cond)
+
+            # handle outputs
+            if tmp_output is not None:
+                output_summary = _compute_summary(tmp_output, percentiles_list)
+            else:
+                output_summary = {i_key: None for i_key in cond_lists.keys()}
+
+            # handle output sensitivities
+            if tmp_output_sensi is not None:
+                output_sensi_summary = _compute_summary(tmp_output_sensi,
+                                                        percentiles_list)
+            else:
+                output_sensi_summary = {i_key: None
+                                        for i_key in cond_lists.keys()}
+
+            # create some PredictionConditionResult to have an easier creation
+            # of PredictionResults for the summaries later on
+            for i_key in cond_lists.keys():
+                cond_lists[i_key].append(
+                    PredictionConditionResult(
+                        timepoints=current_cond.timepoints,
+                        output=output_summary[i_key],
+                        output_sensi=output_sensi_summary[i_key],
+                        observable_ids=current_cond.observable_ids
+                    )
+                )
+
+        self.prediction_summary = {i_key: PredictionResult(
+            conditions=cond_lists[i_key],
+            condition_ids=self.prediction_results[0].condition_ids,
+            comment=str(i_key))
+            for i_key in cond_lists.keys()
+        }
+
+        # also return the object
+        return self.prediction_summary
+
+
+class Collection:
+    """
+    A collection is a thin wrapper around an numpy array.
+    """
+
+    def __init__(self,
+                 x_vectors: np.ndarray,
+                 x_names: Sequence[str] = None,
+                 vector_tags: Sequence[Tuple[int, int]] = None,
+                 coll_type: CollectionType = None,
+                 predictions: Sequence[CollectionPrediction] = None,
+                 lower_bound: np.ndarray = None,
+                 upper_bound: np.ndarray = None):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        x_vectors:
+            parameter vectors of the collection, in the format
+            n_parameter x n_vectors
+        x_names:
+            Names or identifiers of the parameters
+        vector_tags:
+            Additional tag, which adds information about the the parameter
+            vectors of the form (optimization_run, optimization_step) if the
+            collection is created from an optimization result or
+            (sampling_chain, sampling_step) if the collection is created from a
+            sampling result.
```

Comment:
Does this have to be restricted to `(int, int)`? Might be convenient to have more flexibility. But also fine as is.

Reply:
No, doesn't really have to. Many things here are beta version like... That was the most typical example I could think of. The whole ensemble thing is just about to start, much more will come in the future (e.g., create ensembles from pyPESTO/parPE optimization results in an automated fashion). Some of the data members I added try to already account for that, but it's not always easy to anticipate how to best implement things which are yet to come...